### PR TITLE
Add Endace Integration

### PR DESCRIPTION
**Pull request by jamesagarside**
Adding support for Endace. 

The integration contains two inputs, Netflow for use with Endace Flow https://www.endace.com/EndaceFlow and Packbeat Flow Logs for use with Endace vProbe or where systems dont have another way of generating flows externally.

The integration takes information from the flow event and uses it to construct a pivot link which loads the flow in the Endace platform and stores the link under `event.reference`.

### CODE REVIEW

File: packages/endace/data_stream/log/agent/stream/netflow.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+{{/each}}
+{{#contains "forwarded" tags}}
+publisher_pipeline.disable_host: true
+{{/contains}}
+processors:
+- add_fields:
+    target: "_conf"
+    fields:
+      endace_url: {{ endace_url }}
+      test: "test"
```
Comment (mjwolf) [inline packages/endace/data_stream/log/agent/stream/netflow.yml.hbs:39]: You probably don't need this?

File: packages/endace/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      field: 
+        - _tmp_
+      ignore_missing: true
+  - pipeline:
+      name: '{{ IngestPipeline "endace-netflow" }}'
+      tag: pipeline_processor
+      if: ctx.source?.ip != null && ctx.destination?.ip != null && ctx.source?.ip != '0.0.0.0' && ctx.destination?.ip != '0.0.0.0'
+  - remove:
+      field: "_conf"
+on_failure:
```
Comment (mjwolf) [inline packages/endace/data_stream/log/elasticsearch/ingest_pipeline/default.yml:115]: You could use the same `on_failure` section as `packages/endace/data_stream/flow/elasticsearch/ingest_pipeline/default.yml`, and add tags to all processors here. That makes debugging pipeline problems easier

File: packages/endace/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            default: false
+      - type: netflow
+        title: Collect Endace Flow logs
+        description: Collecting Endace Flow logs using the netflow input
+agent:
+  privileges:
+    root: true
+owner:
+  github: elastic/sec-deployment-and-devices
+  type: partner
```
[the remark this answers is not in the document]
Comment (jamesagarside) [reply packages/endace/manifest.yml:99]: As part of this integration the partner has agreed to maintain this

File: packages/endace/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (jamesagarside) [reply packages/endace/data_stream/log/elasticsearch/ingest_pipeline/default.yml:115]: Fixed, thank you :)

File: packages/endace/data_stream/log/agent/stream/netflow.yml.hbs (hunk 1 above)
Comment (jamesagarside) [reply packages/endace/data_stream/log/agent/stream/netflow.yml.hbs:39]: Ahh good spot here, I use it to add the settings to each doc but the `test` bit is futile. I was actually missing a few other settings.

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -152,6 +152,7 @@
 /packages/elastic_agent @elastic/elastic-agent
 /packages/elastic_package_registry @elastic/ecosystem
 /packages/elasticsearch @elastic/stack-monitoring
+/packages/endace @elastic/sec-deployment-and-devices
```
Comment (pkoutsovasilis) [inline .github/CODEOWNERS:155]: I am not entirely sure that @elastic/sec-deployment-and-devices is the whole owner of this integration. One of the streams of this integration, namely `flow`, uses packetbeat and the [owner of this input](https://github.com/elastic/beats/blob/main/.github/CODEOWNERS#L102) is @elastic/sec-linux-platform team. So I think we need to define the ownership of each data stream accordingly (for example this is done for [azure integration](https://github.com/elastic/integrations/blob/main/.github/CODEOWNERS#L72). And of course this data stream has to be reviewed by this team

File: packages/endace/data_stream/log/elasticsearch/ingest_pipeline/endace-netflow.yml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      field: _conf.ip_conv
+      value: "ip_conv={{ source.ip }}%26{{ destination.ip }}"
+      if: ctx.destination?.ip != null && ctx.destination?.ip != '' && ctx.source?.ip != null && ctx.source?.ip != ''
+      tag: endace conversation set
+
+  - set: 
+      description: "Set IP Conversation if either destination.ip and source.ip are present"
+      field: _conf.ipconv
+      value: "ip={{ source.ip }}{{ destination.ip }}"
+      if: ctx._conf.ipconv != null && ctx.destination?.ip != null && ctx.destination?.ip != '' || ctx.source?.ip != null && ctx.source?.ip != ''
```
Comment (pkoutsovasilis) [inline packages/endace/data_stream/log/elasticsearch/ingest_pipeline/endace-netflow.yml:15]: do we need any parenthesis here to group up logical operators?!

File: packages/endace/data_stream/log/elasticsearch/ingest_pipeline/endace-netflow.yml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  - script:
+        source: "ctx._conf.event.end = ctx._conf.event.end - ctx._conf.timedelta"
+        tag: "Calculate Endtime + additional lookback"
+        description: "Calculate Endtime + additional lookback"
+        if: ctx._conf?.event?.end != null && ctx._conf?.event?.end != '' && ctx._conf?.timedelta != null && ctx._conf?.timedelta != ''
+
+
+  - set:
+      field: event.reference
+      value: "{{ _conf.endace_url}}/vision2/pivotintovision/?title=endace_pivot&datasources={{_conf.endace_datasources }}&start={{ _conf.event.start }}&end={{ _conf.event.end }}&tools={{ _conf.endace_tools }}&{{ _conf.ip_conv }}"
```
Comment (pkoutsovasilis) [inline packages/endace/data_stream/log/elasticsearch/ingest_pipeline/endace-netflow.yml:66]: if `ip_conv` is not set is the ending `&` gonna cause any issues?

File: packages/endace/data_stream/log/fields/ecs.yml [hunk 7]
```diff
@@ -0,0 +1,712 @@
...
+- external: ecs
+  name: event.risk_score_norm
+- external: ecs
+  name: event.sequence
+- external: ecs
+  name: event.severity
+- external: ecs
+  name: event.start
+- external: ecs
+  name: event.timezone
```
Comment (pkoutsovasilis) [inline packages/endace/data_stream/log/fields/ecs.yml:235]: if I am not mistaken we need here the `event.reference` as well, right?

File: .github/CODEOWNERS (hunk 4 above)
Comment (jamesagarside) [reply .github/CODEOWNERS:155]: Who is the best person to make this decision?

**Comment by jamesagarside** [comment]
@taylor-swanson @pkoutsovasilis 
Thank you both for reviewing this. Im happy to fix these here but its worth noting the pipelines other than `endace.yml` we're copied from the following Packages. 
https://github.com/elastic/integrations/tree/main/packages/network_traffic/data_stream/flow/elasticsearch/ingest_pipeline
https://github.com/elastic/integrations/tree/main/packages/netflow/data_stream/log/elasticsearch/ingest_pipeline
These changes will need to be reflected here to ensure parity.

File: packages/endace/data_stream/log/elasticsearch/ingest_pipeline/endace-netflow.yml (hunk 6 above)
Comment (jamesagarside) [reply packages/endace/data_stream/log/elasticsearch/ingest_pipeline/endace-netflow.yml:73]: For data to be sent to this pipeline either `source.ip` or `destination.ip` must be set

**Comment by jamesagarside** [comment]
ER for this PR Issue #9579

**Comment by jamesagarside** [comment]
There is a bug in Kibana which means when rendering the `event.reference` pivot field ampersands are escaped which causes issues with the Endace appliance. Two Kibana issues have been raised to fix this behaviour. 
https://github.com/elastic/kibana/issues/188975
https://github.com/elastic/kibana/issues/188973

**Review by mjwolf — APPROVED** [review]

**Comment by taylor-swanson** [comment]
Adding @elastic/sec-linux-platform team as they will be the owners of the `flow` data stream.

**Review by taylor-swanson — COMMENTED** [review]
Few other comments, mostly nit ones.

The bigger concern I had was around the ip and network.type handling in the log data stream. Checking that a v4 address is null or `0.0.0.0` may help that situation, but not sure we if can assume that an ip of `0.0.0.0` is considered an "unset" ip in every circumstance.

File: packages/endace/_dev/build/docs/README.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
+# Endace
+
+Endace is a company known for its network recording, traffic capture, and analysis technology. Endace's solutions are often used for network security, performance monitoring, and troubleshooting.
+This integration allows users to ingest Network flow data from either Endace Flow via syslog input or use Elastic Agent to generate and ship Network Flow data to an Elastic deployment. Both of these methods add the `event.reference` field to each event when ingested into Elasticsearch which is a URL used to pivot to Endace.
```
[the remark this answers is not in the document]
Comment (jamesagarside) [reply packages/endace/_dev/build/docs/README.md:4]: The Netflow integration leverages the Syslog input under the hood I thought

File: packages/endace/data_stream/log/_dev/test/pipeline/test-netflow-log-events.json-expected.json [hunk 9]
```diff
@@ -0,0 +1,3594 @@
...
+                "ip": [
+                    "0.0.0.0"
+                ]
+            },
+            "source": {
+                "ip": "0.0.0.0",
+                "locality": "internal",
+                "port": 136
+            }
+        },
```
[the remark this answers is not in the document]
Comment (jamesagarside) [reply packages/endace/data_stream/log/_dev/test/pipeline/test-netflow-log-events.json-expected.json:3514]: Could we add IPV6 support at a later date? 
We could add a check for type on the default pipeline.

File: packages/endace/manifest.yml [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+name: endace
+title: "Endace"
+version: 0.1.0
+source:
+  license: "Elastic-2.0"
+description: "This Endace integration configures Network Packet Capture for flow generation and adds a pivot field to your Endace platform."
+type: integration
+categories:
+  - aws
+  - cloud
```
[the remark this answers is not in the document]
Comment (jamesagarside) [reply packages/endace/manifest.yml:11]: Endace have a probe which runs in AWS hence these tags. Forgive me if they are reserved for AWS & Cloud native integrations respectively.

[the remark this answers is not in the document]
Comment (taylor-swanson) [reply packages/endace/manifest.yml:11]: I think that's fair, then. Mainly was checking to make sure it wasn't a copy/paste oops.

File: packages/endace/_dev/build/docs/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (taylor-swanson) [reply packages/endace/_dev/build/docs/README.md:4]: They are completely different technologies (Netflow is a binary protocol)

**Pull request closed without merging**